flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -704,12 +882,9 @@ public void receiveDeclineMessage(DeclineCheckpoint message, String taskManagerL
 				if (message.getReason() == null) {
 					checkpointException =
 						new CheckpointException(CheckpointFailureReason.CHECKPOINT_DECLINED);
-				} else if (message.getReason() instanceof CheckpointException) {
-					checkpointException = (CheckpointException) message.getReason();
```

Comment:
I guess this is no longer needed and that's why it was removed?

Reply:
I unified this with the "else" below through `getCheckpointException`.

Reply:
Ops, right. Sorry I missed that :)